bitcoin/script.c

```diff
@@ -521,15 +529,35 @@ u8 **bitcoin_witness_sig_and_element(const tal_t *ctx,
  *     OP_ENDIF
  *     OP_CHECKSIG
  */
-u8 *bitcoin_wscript_to_local(const tal_t *ctx, u16 to_self_delay,
+/* BOLT- #3
+ * ##### Leased channel (`option_will_fund`)
+ * If a `lease` applies to the channel, the `to_local` output of the `accepter`
+ * ensures the `leasor` funds are not spendable until the lease expires.
+ *
+ * In a leased channel, the `to_local` output that pays the `accepter` node
+ * is modified so that its CSV is equal to the greater of the
+ * `to_self_delay` or the `lease_end` - `blockheight`.
+ *
+ *  OP_IF
+ *      # Penalty transaction
+ *      <revocationpubkey>
+ *  OP_ELSE
+ *      MAX(`to_self_delay`, `lease_end` - `blockheight`)
+ *      OP_CHECKSEQUENCEVERIFY
+ *      OP_DROP
+ *      <local_delayedpubkey>
+ *  OP_ENDIF
+ *  OP_CHECKSIG
+ */
+u8 *bitcoin_wscript_to_local(const tal_t *ctx, u16 to_self_delay, u32 csv,
```

Comment:
csv is a terrible name for this parameter!  lease_remaining?

Reply:
updated!